packages/@okta/vuepress-site/docs/guides/configure-native-sso/main/index.md

```diff
@@ -0,0 +1,403 @@
+  --data-urlencode 'audience={audience}'
+```
+
+Note the parameters that are being passed:
+
+* `client_id`: Identifies the new client (for example, client #2) and matches the Client ID of the OAuth 2.0 application that you created.
+* `grant_type`: Identifies the mechanism that Okta uses to authorize the creation of the tokens. Value: `urn:ietf:params:oauth:grant-type:token-exchange`
+* `actor_token`: Identifies the `device_secret` that was returned in response to the authorization code request.
+* `actor_token_type`: Identifies the type of token in the `actor_token` parameter.
+* `subject_token`: Identifies the `id_token` that was returned in response to the authorization code request.
+* `subject_token_type`: Identifies the type of token in the `subject_token` parameter. Value: `urn:ietf:params:oauth:token-type:id_token`
+* `scope`: The scopes that you want to be included in the access token.
+* `audience`: Identifies the audience of the authorization server. Locate the value for the `audience` parameter in the Admin Console. From the left navigation pane, select **Security** > **API**. On the **Authorization Server** tab, select the pencil icon for the authorization server that you need the `audience` value for.
+
+If the request is successful, the response returned includes the following tokens: `id_token`, `refresh_token`, and `access token`.
+
+> **Note:** You can pass an expired ID token as part of the token exchange grant as long as the `device_secret` (`sid`) that the `id_token` is associated with is still valid.
+
+**Response example**
+
+```JSON
+{
+   "token_type": "Bearer",
+   "expires_in": 3600,
+   "access_token": "eyJraW...TbvY9A",
+   "scope": "openid",
+   "id_token": "eyJraW...5aCZrA",
+   "issued_token_type": "urn:ietf:params:oauth:token-type:access_token"
+}
+```
+
+## Validate the device secret
+
+Occasionally you may want to verify that the device secret is still valid by using the `/introspect` endpoint.
+
+**Example introspect request**
+
+```bash
+  curl --request POST \
+  --url https://${yourOktaDomain}/oauth2/v1/introspect \
+  --header 'Accept: application/json' \
+  --header 'Content-Type: application/x-www-form-urlencoded' \
+  --data-urlencode 'client_id={client #1 id}' \
+  --data-urlencode 'token={device_secret}' \
+  --data-urlencode 'token_type_hint={device_secret}'
+```
+
+The `/introspect` endpoint returns the `sid` that it's tied to. The same value is present in the ID token. By doing this, you can correlate and identify the ID tokens that are minted with the same device secret.
+
+## Revoke the device secret to end a desktop session
+
+Sometimes you have to end a user's desktop session. When you do that, you are signing the user out of one app as well as all of the other registered apps. To end a desktop session, you must revoke the device secret. The revoke request signs the user out from all of the apps that are a part of the Native SSO flow.
+
+**Example request**
```

Comment:
Why is there no response example for device secret?
Example:
{
    "active": true,
    "sid": "1020ci3hNawShud5mek1S4ahw",
    "token_type": "urn:x-oath:params:oauth:token-type:device-secret"
}

Reply:
This is for the introspection endpoint response